src/Avalonia.Controls/VirtualizingStackPanel.cs

```diff
@@ -74,11 +93,35 @@ public class VirtualizingStackPanel : VirtualizingPanel, IScrollSnapPointsInfo
         private Control? _realizingElement;
         private int _realizingIndex = -1;
 
+        /// <summary>
+        /// The factor to determine additional space to maintain above and below the viewport.
+        /// </summary>
+        private double _bufferFactor; 
+        
+        private bool _hasReachedStart = false;
+        private bool _hasReachedEnd = false;
+        private bool _cacheElementMeasurements = false;
+        private Size _previousSize;
+        private readonly HashSet<Control> _unchangedElements = new();
+        private readonly bool _traceVirtualization = false;
```

Comment:
Before we merge this PR, we should remove the debug code - I can understand that it's useful for debugging, but we don't have debug code lying around anywhere else. Hopefully it should all be covered by unit tests by now.

Reply:
absolutely - as mentioned in my PR, i just kept this for you to follow along more easily